src/node_crypto.cc

```diff
@@ -5263,7 +5263,7 @@ void PBKDF2(const FunctionCallbackInfo<Value>& args) {
   THROW_AND_RETURN_IF_NOT_BUFFER(args[1], "Salt");
 
   pass = static_cast<char*>(node::Malloc(passlen));
-  if (pass == nullptr) {
+  if (pass == nullptr && passlen > 0) {
     FatalError("node::PBKDF2()", "Out of Memory");
   }
```

Comment:
@mscdex Thanks for your detailed explanation, after I did read [src/util-inl.h#L232-L245](https://github.com/nodejs/node/blob/master/src/util-inl.h#L232-L245), I would think should we actually return a specific unique pointer for zero-sized allocation requests, thus we don't need any changes on our previous check by `nullptr`.

Introducing an extra normalization to return `nullptr` is creating a conflicts with our previous changes.